e2e/ctl_v3_alarm_test.go

```diff
@@ -99,6 +108,101 @@ func alarmTest(cx ctlCtx) {
 	}
 }
 
+type fakeConsistentIndex struct{ rev uint64 }
+
+func (f *fakeConsistentIndex) ConsistentIndex() uint64 { return f.rev }
+
+func alarmCorruptTest(cx ctlCtx) {
+	for i := 0; i < 10; i++ {
+		if err := ctlV3Put(cx, "k", "v", ""); err != nil {
+			if cx.dialTimeout > 0 && !isGRPCTimedout(err) {
+				cx.t.Fatalf("putTest ctlV3Put error (%v)", err)
+			}
+		}
+	}
+	cx.epc.procs[0].Stop()
+
+	// Corrupt member 0 by modifying backend offline.
+	fp := filepath.Join(cx.epc.procs[0].Config().dataDirPath, "member", "snap", "db")
+	be := backend.NewDefaultBackend(fp)
+	s := mvcc.NewStore(be, nil, &fakeConsistentIndex{13})
+	s.Put([]byte("abc"), []byte("def"), 0)
+	s.Put([]byte("xyz"), []byte("123"), 0)
+	s.Compact(5)
+	s.Commit()
+	s.Close()
+	be.Close()
+
+	eps := cx.epc.EndpointsV3()
+
+	// Wait for cluster so Puts succeed in case member 0 was the leader.
+	cli1, err := clientv3.New(clientv3.Config{Endpoints: []string{eps[1]}, DialTimeout: 3 * time.Second})
+	if err != nil {
+		cx.t.Fatal(err)
+	}
+	defer cli1.Close()
+
+	if _, err = cli1.Get(context.TODO(), "k"); err != nil {
+		cx.t.Fatal(err)
+	}
+	cli1.Put(context.TODO(), "xyz", "321")
+	cli1.Put(context.TODO(), "abc", "fed")
+
+	// Restart with corruption checking
+	cx.epc.procs[1].Stop()
+	cx.epc.procs[2].Stop()
+	cx.epc.Restart()
+
+	// wait until corruption detected
+	cli0, err := clientv3.New(clientv3.Config{Endpoints: []string{eps[0]}, DialTimeout: 3 * time.Second})
+	if err != nil {
+		cx.t.Fatal(err)
+	}
+	defer cli0.Close()
+
+	sresp, err := cli1.Status(context.TODO(), eps[0])
+	if err != nil {
+		cx.t.Fatal(err)
+	}
+	id0 := sresp.Header.GetMemberId()
+
+	corrupted := false
+	for i := 0; i < 5; i++ {
+		presp, perr := cli0.Put(context.TODO(), "abc", "aaa")
```

Comment:
the boot check should prevent member 0 from servicing any KV RPCs, otherwise it doesn't provide any better guarantees than the periodic corruption check